### DISCUSSION TITLE
Plasmic

### DISCUSSION THREAD

**Discussion by holden** [swellstores/community#142] [Help] [answered]
Quick question. We just started building our frontend template and have started off using Plasmic.app. One thing we noticed and maybe this was intentional, but is the Origin template on Plasmic not supposed to be feature complete?

Eg we start developing with it and noticed many areas of the template are not setup by default.  Is it just supposed to be this basic or did we miss something?

For example the shop page doesn't load anything by default: https://cirqus.plasmic.run/shop

I would just like to be sure we didn't set up something incorrectly before we start trying to make all the pages work, etc.

Thanks,

**Comment by sags95** [comment]
One thing to check is if the Swell Provider configuration is correct in Project Settings. The default `storeId` and `publicKey` would have to be changed in order to pull product and category info from your store.

**Reply by holden** [reply]
Yes, I did this. It pulls some of the information, you can see it's pulling our products on some of the pages. It just feels like the template is not 100% feature complete as the full origin template. It has most of the general pages there but not necessarily all the components work. I was just wondering if this is how it is supposed to be, as I just expected it would work the same as if we took the Origin template from Github.

**Reply by sags95** [reply]
The Plasmic team developed and maintains the template rather than Swell, so I would have to defer to their team regarding the overall functionality of the template.

**Reply by yang** [reply]
Hi @holden , I'm from the Plasmic team, happy to take a look at this. Would you have a pointer to the Plasmic project URL?

**Reply by holden** [reply]
Hi @yang, sure, that'd be great. The site we're presently working on is here: https://cirqus.plasmic.run/

But I also just setup a new origin template from scratch here: https://cirqus-origin-vanilla.plasmic.run/shop which you can compare to the Swell default here: https://cirqus.swell.store/ both using the same data and the same template Origin (from today). I think it's just that many of the pages are missing. 

We've started adding the pages manually, which has helped up learn Plasmic.  I was just surprised when we started as I expected that the Origin template (on plasmic) would have more to features by default.  Unless I'm missing something?

Thanks for your help.

**Reply by yang** [reply, marked as answer]
Hi @holden, I think I misunderstood the original issue report. You're right, that template in Plasmic is a subset of the full template from Swell—its main goal was to demo how all the Swell commerce features work in the Plasmic integration. The other parts are mostly more of the same (displaying product catalog data in various designs, e.g. in the PDP and nav menu). The main part that would require custom code at the moment is the cart sidebar. The Plasmic project is meant to be overlaid on top of the base Next.js Swell commerce codebase, so you'd already have the sidebar UX there.

But it's definitely confusing what the state is. We're relabeling the project template to be more clear for now.

[Let us know](https://plasmic.app/slack) if there are other Plasmic questions we can help with.

**Reply by yang** [reply]
Just went ahead and updated the Origin template with working PDP, if that helps.
